src/clients/HttpTunneler.cc

```diff
@@ -341,6 +338,7 @@ Http::Tunneler::handleResponse(const bool eof)
     // preserve any bytes sent by the server after the CONNECT response
     futureAnswer.leftovers = readBuf;
 
+    debugs(11, DBG_PROTOCOL, "switch to TCP on:" << connection);
```

Comment:
Please remove this line. This change is controversial (e.g., I am not sure we are switching to TCP here and would need to develop a clear notion of protocol scope to be certain) and lies outside the declared PR scope.

Reply:
The "protocol scope" you are looking for is defined in RFC 9111 section 9.3.6:
 "if successful, thereafter restrict its behavior to blind forwarding of data, in both directions, until the tunnel is closed."

Squid currently only supports (and HttpTunneler implements) HTTP/1.x tunnels over TCP transport layer. So from the perspective of HTTP protocol trace this is a switch to TCP.

FTR, this line does **not** relate to whether the opaque "TCP" immediately switches to some other Foo protocol (eg TLS for SSL-Bump). That would be shown in the trace of TCP protocol as a "switch to Foo" log entry.